### ISSUE TITLE
ignoreWhen (was: reject)

### ISSUE THREAD

**Issue by tailec** [RxSwiftCommunity/RxSwift-Ext#1]
### reject
`reject` operator as an opposite of `filter`.

### Motivation for inclusion
It's trivial operator but I find it more verbose than i.e. `.filter { $0 != foo }`.

### Example of usage
```
Observable.just("Act 1 Scene 5: The Great Hall in Capulet's mansion")
  .reject { $0.hasPrefix("Act 1") }
```
Instead of

```
Observable.just("Act 1 Scene 5: The Great Hall in Capulet's mansion")
  .filter { !$0.hasPrefix("Act 1") }
```

**Comment by bontoJR** [comment]
@tailec I create a quick template for issues, can you try to re-structure this issue with the template I proposed? So we can see if it makes sense?

**Comment by tailec** [comment]
@bontoJR fixed :)

**Comment by fpillet** [comment]
@tailec: shouldn't this operator be called `rejectIf` since the closure is a conditional ?

**Comment by tailec** [comment]
@fpillet Yeah, I personally like `rejectWhen`.

**Comment by fpillet** [comment]
In fact I think even the term `reject` is too strong. RxSwift has a `ignoreElements()` operator. Lets name yours `ignoreWhen` to make it clear that it won't trigger an error, just ignore elements

**Comment by fpillet** [comment]
Let's go ahead with this one. I checked the operators list on [ReactiveX](http://reactivex.io/documentation/operators.html) and couldn't come up with a better name. I think we're good.

**Comment by fpillet** [comment]
implemented by c9eceea
